core/src/databases/database.rs

```diff
@@ -158,16 +141,16 @@ impl Database {
                 ));
 
                 let insert_execute_start = utils::now();
-                rows_by_table
-                    .iter()
+                rows.iter()
                     .filter(|(_, rows)| !rows.is_empty())
-                    .map(|(table_name, rows)| {
-                        let table_schema = match schema.get(table_name) {
-                            Some(s) => Ok(s),
-                            None => Err(anyhow!("No schema found for table {}", table_name)),
-                        }?;
-
-                        let (sql, field_names) = table_schema.schema.get_insert_sql(table_name);
+                    .map(|(table, rows)| {
+                        let table_schema =
+                            match schema.iter().find(|s| s.table.name() == table.name()) {
```

Comment:
Can we create a map first 😬 ? might be better to avoid O(n*m) complexity (even if number of tables should stay relatively small)

Reply:
with m << n and small :)

Reply:
mmh OK fair

Reply:
O(n + m^2) to be fair but m tiny :-)

Reply:
Saving this thread for the day `m` is no longer tiny !